ground/src/main/java/com/google/android/ground/repository/MutationRepository.kt

```diff
@@ -76,22 +76,27 @@ constructor(
    * Returns all LOI and submission mutations in the local mutation queue relating to LOI with the
    * specified id.
    */
-  suspend fun getPendingMutations(locationOfInterestId: String): List<Mutation> {
-    val pendingLoiMutations =
-      localLocationOfInterestStore
-        .findByLocationOfInterestId(locationOfInterestId, MutationEntitySyncStatus.PENDING)
-        .map { it.toModelObject() }
-    val pendingSubmissionMutations =
-      localSubmissionStore
-        .findByLocationOfInterestId(locationOfInterestId, MutationEntitySyncStatus.PENDING)
-        .mapNotNull { entity ->
-          localSurveyStore.getSurveyByIdSuspend(entity.surveyId)?.let { entity.toModelObject(it) }
-        }
-    return pendingLoiMutations + pendingSubmissionMutations
+  suspend fun getMutations(
+    loidId: String,
+    entitySyncStatus: MutationEntitySyncStatus
+  ): List<Mutation> {
+    val loiMutations =
+      localLocationOfInterestStore.findByLocationOfInterestId(loidId, entitySyncStatus).map {
+        it.toModelObject()
```

Comment:
Does `map(it::toModelObject)` also work?

Reply:
`it` only works in SAM interfaces `{ }`

In this case, `.map(LocationOfInterestMutationEntity::toModelObject)` works.

Reply:
Fixed